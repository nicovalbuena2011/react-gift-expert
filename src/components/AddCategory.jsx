import { useState } from "react"
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) =>{
    //console.log(onNewCategory); //Recibe la funcion del otro componente

    //Utilizamos el hook useState para verificar en todo momento el estado que es lo que ingresa el ururario y lo mosifica
    const [ inputValue, setInputValue ] = useState(' ');

    const onInputChange = (event) =>{

        //console.log( event.target.value );
        setInputValue( event.target.value );
    }

    const onSubmit = (event) =>{
        event.preventDefault();


        const value = inputValue.trim();

        if (value.length <= 1) return;

        onNewCategory( value );
        setInputValue('');

    }

    //console.log(categories);


    return(
        <form onSubmit={ (event) => onSubmit(event) } >
            <input
                type="text"
                placeholder="Buscar Gifts"
                value={ inputValue }
                onChange={ onInputChange }
            
            />
        </form>
    )

}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
    
}