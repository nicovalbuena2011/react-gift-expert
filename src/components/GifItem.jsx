import PropTypes from 'prop-types';

export const GiftItem = ({ id, title, url }) =>{

    return(
        <div className="card">
            <img src={ url } alt= { title } />
            <p> {title} </p>
          

        </div>
    )
}

GiftItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

// GiftItem.defaultProps = {
//     tittle: "No hay titulo",
//     url: "No encontrado",
    
// }