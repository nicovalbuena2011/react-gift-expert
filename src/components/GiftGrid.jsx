import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";
import { GiftItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";


export const GiftGrid = ({ category }) =>{

    //console.log( category );
    //Custom hook
    const { images, isLoading } = useFetchGifs( category ); 

    return(
        <>
            <h3>{ category }</h3>

            {
                isLoading && ( <h2> Cargando </h2> )
            }

            <div className="card-grid">
                {   
                    images.map( ( images ) =>  (           
                        <GiftItem 
                            key = {images.id}
                            {...images} //Le manda toda la informacion de las images en el useState.
                        />                      
                    ))
                }       
            </div>
         
        </>
    )


}