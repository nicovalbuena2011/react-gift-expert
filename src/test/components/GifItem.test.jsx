import {fireEvent, render, screen } from "@testing-library/react";
import { GiftItem } from '../../components/GifItem'

describe('Prueba en el componente GifItem ', () =>{

    const titulo = 'Naruto';
    const url = 'https://localhost/url';

    test('Debe tener el titulo y la url obligatoria. se compara con el snapshot ', ()=>{

        const { container  } = render( <GiftItem title = { titulo } url = {url} /> )
        expect( container ).toMatchSnapshot();
        

    });

    test('Debe de mostrar la imagen con el url y el alt indicado', ()=>{
        render( <GiftItem title = { titulo } url = {url} /> )
        //screen.debug(); //Muestra el componente renderizados
        const { src, alt } = screen.getByRole('img'); //Seleccionamos la imagen
        //console.log(src,alt);
        expect( src ).toBe( url );
        expect( alt ).toBe( titulo );


    });


});