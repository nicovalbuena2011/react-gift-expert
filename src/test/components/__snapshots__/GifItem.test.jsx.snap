// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en el componente GifItem  Debe tener el titulo y la url obligatoria. se compara con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Naruto"
      src="https://localhost/url"
    />
    <p>
       
      Naruto
       
    </p>
  </div>
</div>
`;
